Create documentation for bar chart (#1246)
timescale/docs: timescaledb/tutorials/grafana/visualizations/bar-chart.md

  with pre-aggregated data.

There are a few different kinds of bar charts to choose from, including
vertical, horizontal, and stacked bar charts. All of these types are covered in
this tutorial.

## Prerequisites

<GrafanaVizPrereqs />

## Create a bar chart with preaggregated data

Create a bar chart visualization using the data in the table `stocks_real_time`.

<procedure>

### Creating a bar chat with preaggregated data

1. In the query editor, use this SQL to query a bar chart dataset. Use the
    variable `$bucket interval` for the time period covered by the bar chart:

    ```sql
    SELECT
        time_bucket('$bucket_interval', time) AS time,
        symbol,
        AVG(price) as price
    FROM stocks_real_time srt
    WHERE symbol = $symbol
        AND time >= $__timeFrom()::timestamptz AND time < $__timeTo()::timestamptz
    GROUP BY time_bucket('$bucket_interval', time), symbol;
    ```

1. In the Grafana dashboard, in the `Dashboard variable` field, select a stock
    to graph. Adjust the time range of the dashboard if needed. Make sure the

>     returned data has column named `time` that contains timestamps. The
>     timestamps should be in ascending order, you might get errors otherwise. The

Reviewer: ```suggestion
    returned data has a column named `time` that contains timestamps. The
    timestamps should be in ascending order. Otherwise, you get an error. The
```

    returned data looks like this:

>     <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/Tabledataforgoogle.png" alt="Screenshot of the table view of valid time series data for google stock."/>

Reviewer: ```suggestion
    <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/Tabledataforgoogle.png" alt="Screenshot of the table view of valid time-series data for Google stock."/>
```

1. In the `Visualizations` field, select `Bar chart`. Grafana turns the query
    into a bar chart. This example shows a vertical bar chart price distribution
    of Google stock, which ranges between $2836 and $2108 within a specific
    period:

    <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/googlebarchart.png" alt="Screenshot of the vertical bar chart produced by Grafana. The vertical bar chart represents the price of Google in the past 2 months."/>

1. You can convert from a vertical bar chart to a horizontal one to get room
    for a longer label along the vertical axis. In the dashboard, navigate to
    the bar chart section. In the `Orientaton` section, click `horizontal`. The
    horizontal bar chart looks like this:

>     <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/googlebarchart.png" alt="Screenshot of the horizontal bar chart produced by Grafana. The horizontal bar chart represents the price of google in the past 2 months."/>

Reviewer: ```suggestion
    <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/googlebarchart.png" alt="Screenshot of the horizontal bar chart produced by Grafana. The horizontal bar chart represents the price of Google in the past 2 months."/>
```

</procedure>

## Create multiple bar charts

> If you want compare the distributions of 4 or more different stocks, you can

Reviewer: ```suggestion
If you want to compare the distributions of 4 or more different stocks, you can
```

create a panel containing multiple bar charts. The database returns the
transactions for all selected values, and Grafana buckets them in separate bar
charts.

<procedure>

> ### Creating multiple bar chart in a single panel

Reviewer: ```suggestion
### Creating multiple bar charts in a single panel
```

  And the displayed graph looks like this:

  <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/multiplebarchart.png" alt="Screenshot of the multiple bar chart produced by Grafana. The multiple bar chart represents the price of four different stocks in the past 1 month."/>

1. In the graph you just created, you can see the 5 different stocks, but it is
   difficult to differentiate them. To tell them apart, you can adjust the color
   of each stock by clicking the legend to the left of each line, and picking a
   color for each bar:

  <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/visualizations/barchart/multicoloredbarchart.png" alt="Screenshot of Grafana plot, showing 5 bar chart of stock values in green, blue, red, purple and orange."/>

</procedure>

## Create a stacked bar chart

You can use a stacked bar chart to show how different stock divides into smaller

> categories, and what the relationship of each part has on the total amount.

Reviewer: ```suggestion
categories, and what impact each part has on the total amount.
```

The previous examples used vertical, horizontal, and multiple bar charts with
price transactions. In this section, you see the trading volume of each stock
using a bucket interval.

The `stock_real_time` hypertable contains a column with the daily cumulative
traded volume. This helps to calculate the volume of data for each bucket.

<procedure>

### Creating a stacked bar chart

1. At the psql prompt, update the earlier query to find the maximum `day_volume`
   value for a symbol within a bucket. Then, subtract each maximum from the
   previous bucket's maximum. The difference gives the traded volume for that
   bucket:

   ```sql
      SELECT
          time_bucket('$bucket_interval', time) AS time,
          symbol,
          MAX(day_volume) - LAG(max(day_volume), 1) OVER(

>           PARTITION BY symbol
>           ORDER BY time_bucket('$bucket_interval', time)

Reviewer: ```suggestion
              PARTITION BY symbol
              ORDER BY time_bucket('$bucket_interval', time)
```
